data/generate_constants.rb

```diff
@@ -0,0 +1,231 @@
+require_relative '../lib/unicode/emoji/constants'
+require_relative '../lib/unicode/emoji/index'
+require_relative '../lib/unicode/emoji/lazy_constants'
+
+include Unicode::Emoji
+
+def write_regexes(regexes, dirpath)
+  regexes.each do |const_name, regex|
+    write_regex(const_name, regex, dirpath)
+  end
+end
+
+def write_regex(const_name, regex, dirpath)
+  filename = const_name.to_s.downcase
+  filepath = File.join(dirpath, "#{filename}.rb")
+
+  File.write(filepath, <<~CONTENT)
+    # This file was generated. Please, do not edit this file by hand.
+    module Unicode
+      module Emoji
+        #{const_name} = #{regex.inspect}
+      end
+    end
+  CONTENT
+  puts "#{const_name} written to #{filepath}"
+end
+
+# Converts [1, 2, 3, 5, 6, 20, 21, 22, 23, 100] (it does not need to be sorted) to [[1, 2, 3], [5, 6], [20, 21, 22, 23], [100]]
+def groupify(arr)
+  arr = arr.sort
+  prev = nil
+  arr.slice_before do |el|
+    (prev.nil? || el != prev + 1).tap { prev = el }
+  end
+end
+
+# Converts [1, 2, 3, 5, 6, 20, 21, 22, 23, 100] (it does not need to be sorted) to [1..3, 5, 6, 20..23, 100]
+def rangify(arr)
+  groupify(arr).map do |group|
+    group.size < 3 ? group : Range.new(group.first, group.last)
+  end.flatten
+end
+
+def pack(ord)
+  Regexp.escape(Array(ord).pack("U*"))
+end
+
+def join(*strings)
+  "(?:" + strings.join("|") + ")"
+end
+
+def character_class(range)
+  "[" + pack(range.first) + "-" + pack(range.last) + "]"
+end
+
+def pack_and_join(ords)
+  if ords.any? { |e| e.is_a?(Array) }
+    join(*ords.map { |ord| pack(ord) })
+  else
+    join(rangify(ords).map do |el|
+      el.is_a?(Range) ? character_class(el) : pack(el)
+    end)
+  end
+end
```

Comment:
Not sure if it has any (positive) performance/memory implications, but, building upon the character class approach, the regex will get slightly cleaner, if we use a single character class:

```suggestion
def character_class(ords_with_ranges)
  "[" + ords_with_ranges.map{ |ord_or_range|
      ord_or_range.is_a?(Range) ?
        pack(ord_or_range.first) + "-" + pack(ord_or_range.last) :
        pack(ord_or_range)
    }.join +
  "]"
end

def pack_and_join(ords)
  if ords.any? { |e| e.is_a?(Array) }
    join(*ords.map { |ord| pack(ord) })
  else
    character_class(rangify(ords))
  end
end
```